sys/net/application_layer/dhcpv6/client.c

```diff
@@ -215,7 +213,20 @@ static inline bool _is_tid(dhcpv6_msg_t *msg)
 
 static inline uint32_t _now_cs(void)
 {
+#if IS_USED(MODULE_ZTIMER)
+    return (uint32_t)(ztimer_now(ZTIMER_MSEC) / MS_PER_CS);
+#else
     return (uint32_t)(xtimer_now_usec64() / US_PER_CS);
+#endif
+}
+
+static inline uint32_t _now_sec(void)
```

Comment:
There is a potential footgun with using both timestamps from ZTIMER_SEC and ZTIMER_MSEC, as both could be provided by different hardware. One could (or at least I can see myself doing so) assume that the timestamps can be converted to each other.

I don't see this incorrect assumption anywhere here in the code, but I'm not really familiar with it.

Reply:
Yes, the timestamp should be used independent from each other and are only used for the respective timeouts using the different scales.

Reply:
there are multiple potential foot guns, but none of them is worse than it was before with just xtimer, especialiy within reasnonable limit of leasetime and T1 and T2 I would advice against giving longer than 1 month leases

Reply:
The drift pontential of msec vs sec is low enough for T1 and T2 application,

The problem left is one might uses SEC as some kind of RTC (which should not be done), combine that with DHCP and some NTP after getting the dhcp lease and you get a time scramble, this is a problem of the ztimer and not of this DHCP.